File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.py
def solution(s):
    answer = True
    ps = 0
    ys = 0
    for i in s:
        if i == 'p' or i == 'P':
            ps += 1
        elif i == 'y' or i == 'Y':
            ys += 1
        else:
            answer = True

    if ps == ys:
        answer = True
    else:
        answer = False
    
    return answer

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;

        int pCount = 0;
        int yCount = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == 'P' || s.charAt(i) == 'p') {
                pCount++;
            } else if (s.charAt(i) == 'Y' || s.charAt(i) == 'y') {
                yCount++;
            }
        }
        if (pCount == yCount) {
            answer = true;
        } else if (pCount != yCount) {
            answer = false;
        } else {
            answer = true;
        }

        return answer;
    }
}